### PR TITLE
[SPARK-14038][SQL] enable native view by default

### PR DESCRIPTION
**Pull request by cloud-fan**
## What changes were proposed in this pull request?

As we have completed the `SQLBuilder`, we can safely use it to support native view.
This PR enables native view by default, and removes the old branch that implement native view by just wrapping the view string.
## How was this patch tested?

existing tests.

### CODE REVIEW

**Comment by cloud-fan** [comment]
cc @yhuai @liancheng

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/CreateViewAsSelect.scala [hunk 1]
```diff
@@ -73,15 +68,18 @@ private[hive] case class CreateViewAsSelect(
...
     } else {
-      wrapViewTextWithSelect
+      val projectList = childSchema.zip(tableDesc.schema).map {
+        case (attr, col) => Alias(attr, col.name)()
+      }
+      sqlContext.executePlan(Project(projectList, child)).analyzed
     }
 
-    val viewSchema = {
+    val viewText = new SQLBuilder(logicalPlan, sqlContext).toSQL
```
Comment (liancheng) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/CreateViewAsSelect.scala]: How about catching exceptions thrown from `toSQL` and suggest the user to turn off native view and retry?

File: sql/core/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala [hunk 2]
```diff
@@ -365,21 +365,10 @@ object SQLConf {
...
-  val CANONICAL_NATIVE_VIEW = booleanConf("spark.sql.nativeView.canonical",
     defaultValue = Some(true),
-    doc = "When this option and spark.sql.nativeView are both true, Spark SQL tries to handle " +
-          "CREATE VIEW statement using SQL query string generated from view definition logical " +
-          "plan.  If the logical plan doesn't have a SQL representation, we fallback to the " +
-          "original native view implementation.",
-    isPublic = false)
+    doc = "When true, CREATE VIEW statement will be handled by Spark SQL instead of Hive native " +
+          "command, using SQL query string generated from view definition logical plan.  Note " +
+          "that this feature may break in some corner cases, you can disable it to workaround.")
```
Comment (liancheng) [inline sql/core/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala]: We can remove the last sentence and instead suggest the user to turn it off when `toSQL` fails.

**Comment by liancheng** [comment]
LGTM except for two minor comments.

File: sql/core/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala (hunk 2 above)
Comment (yhuai) [inline sql/core/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala]: Is there any known issue? Disabling it does not really work for data source tables. So, for this kind of cases, setting `spark.sql.nativeView` to true and `spark.sql.nativeView.canonical` to false is the only workaround, right?

**Comment by cloud-fan** [comment]
closing in favor of a safer version: https://github.com/apache/spark/pull/11872

**Pull request closed without merging**